### PR TITLE
Adding new DNS records page design

### PR DESCRIPTION
**Pull request by delputnam**
This PR adds the new DNS Records design behind a feature flag since there are still several other parts that need integration work before we can make this available to users. You can test this by setting the `domains/dns-records-redesign` feature flag.

Also, there are a few design updates to the header buttons, but we will handle those in a separate PR (they were not introduced in this PR).

The new page looks like this:

Screen Shot 2021-11-08 at 7 49 02 AM

On Mobile:

calypso localhost_3000_domains_manage_a8ctest com_dns_a8ctest com(iPhone X)

The action menu at the top of the page has a single option right now: "Restore default A records"
Screen Shot 2021-11-07 at 8 02 09 AM

Screen Shot 2021-11-07 at 8 01 45 AM

Selecting this will show a confirmation notice:

Screen Shot 2021-11-07 at 8 03 29 AM

Once the back-end changes are complete, this option will also show the specific A default A records:

Screen Shot 2021-11-08 at 8 02 37 AM

Standard DNS records have two options on the action menu:

Screen Shot 2021-11-08 at 8 05 34 AM

Currently the "Edit" action does nothing since the Edit page is being done in a separate PR. Once this PR is merged, we'll add another PR to integrate the two.

The Domain Connect records has two different options on the action menu:

Screen Shot 2021-11-08 at 8 09 32 AM

If the record is currently disabled, the options are as follows:

Screen Shot 2021-11-08 at 8 11 01 AM

The "What's this?" option shows a dialog with the info text that was displayed inline on the old design:

Screen Shot 2021-11-07 at 8 03 16 AM

### CODE REVIEW

File: client/my-sites/domains/domain-management/dns/dns-records-list.jsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+				}
+			} );
+
+			return;
+		}
+
+		const successMessage =
+			'delete' === action
+				? translate( 'The DNS record has been deleted.' )
+				: translate( 'The DNS record has been disabled.' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/dns/dns-records-list.jsx:128]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'The DNS record has been deleted.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/dns-records-page-design)

File: client/my-sites/domains/domain-management/dns/dns-records-list.jsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+		const errorMessage =
+			'delete' === action
+				? translate( 'The DNS record has not been deleted.' )
+				: translate( 'The DNS record has not been disabled.' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/dns/dns-records-list.jsx:133]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'The DNS record has not been added.' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/dns-records-page-design)

File: client/my-sites/domains/domain-management/dns/dns-record-data.jsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+				return translate( '%(data)s with priority %(aux)s', {
+					args: {
+						data,
+						aux,
+					},
+				} );
+
+			case 'SRV':
+				return translate(
+					'%(service)s (%(protocol)s) on target %(target)s:%(port)s, ' +
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/dns/dns-record-data.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Service %(service)s (%(protocol)s) on target %(target)s:%(port)s, with priority %(aux)s and weight %(weight)s' )` *ES Score: 9*

File: client/my-sites/domains/domain-management/dns/dns-records-list.jsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+			}
+		);
+	};
+
+	addDns( record ) {
+		const { translate } = this.props;
+
+		this.props.addDns( this.props.selectedDomainName, record ).then(
+			() => {
+				this.props.successNotice( translate( 'The DNS record has been enabled.' ), {
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/dns/dns-records-list.jsx:160]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'The DNS record has been deleted.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/dns-records-page-design)

File: client/my-sites/domains/domain-management/dns/dns-records-list.jsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+					duration: 5000,
+				} );
+			},
+			( error ) => {
+				this.props.errorNotice(
+					error.message || translate( 'The DNS record could not be enabled.' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/dns/dns-records-list.jsx:166]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'The DNS record could not be restored.' )` *ES Score: 8*

File: client/my-sites/domains/domain-management/dns/dns-records.jsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+			</Main>
+		);
+	}
+
+	render() {
+		const { showPlaceholder, selectedDomainName, selectedSite } = this.props;
+
+		return (
+			<Fragment>
+				<QuerySiteDomains siteId={ selectedSite.ID } />
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-records.jsx]: I wonder if we need to query the site's domains here - is this used to get the `pointsToWpcom` property?

File: client/my-sites/domains/domain-management/dns/dns-records-list-item.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+		>
+			<div className="dns-records-list-item">
+				<div className="dns-records-list-item__data dns-records-list-item__type">
+					<strong>{ type }</strong>
+				</div>
+				<div className="dns-records-list-item__data dns-records-list-item__name">
+					<span>{ name }</span>
+				</div>
+				<div className="dns-records-list-item__data dns-records-list-item__value">
+					<span>{ value ?? 'example-value' }</span>
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-records-list-item.jsx]: Should this `example-value` be removed?

File: client/my-sites/domains/domain-management/dns/dns-record-data.jsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		switch ( type ) {
+			case 'MX':
+				return translate( '%(data)s with priority %(aux)s', {
+					args: {
+						data,
+						aux,
+					},
+				} );
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-record-data.jsx]: Maybe we should add some comments explaining the variables for the translators?

**Review by leonardost — APPROVED** [review]
LGTM, thanks for this! Tested and it works and looks very nice. Also thank you for the detailed PR description 🙇‍♂️ 

Left just some minor comments, but nothing blocking.

One last thing I noticed was that a long domain record value can look weird in mobile view:

Annotation on 2021-11-09 at 15-20-48

Maybe we could hide the overflowing part with ellipsis there, or make it wrap to a second line?

File: client/my-sites/domains/domain-management/dns/dns-records-list-item.jsx (hunk 7 above)
Comment (delputnam) [reply client/my-sites/domains/domain-management/dns/dns-records-list-item.jsx]: Yes. 🤦  Thanks.  That was for testing.

File: client/my-sites/domains/domain-management/dns/dns-records.jsx (hunk 6 above)
Comment (delputnam) [reply client/my-sites/domains/domain-management/dns/dns-records.jsx]: Yep...good point. Copy-pasta.  We don't need this one any longer.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6956997

Thank you @delputnam for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6956997).

**Pull request merged**